src/commands.rs

```diff
@@ -2771,6 +2782,24 @@ aborted.
     Ok(())
 }
 
+fn cmd_resolve(
+    ui: &mut Ui,
+    command: &CommandHelper,
+    args: &ResolveArgs,
+) -> Result<(), CommandError> {
+    let mut workspace_command = command.workspace_helper(ui)?;
+    let commit = workspace_command.resolve_single_rev(&args.revision)?;
+    workspace_command.check_rewriteable(&commit)?;
+    let mut tx =
+        workspace_command.start_transaction(&format!("mergetool commit {}", commit.id().hex()));
+    let new_tree_id = workspace_command.run_mergetool(ui, &commit.tree(), &args.path)?;
+    CommitBuilder::for_rewrite_from(ui.settings(), &commit)
+        .set_tree(new_tree_id)
+        .write_to_repo(tx.mut_repo());
+    workspace_command.finish_transaction(ui, tx)?;
+    Ok(())
+}
```

Comment:
Nice! This looks very simple and clear (to me anyway)

Reply:
👍 Thanks!